2020 Indiana State House/Senate Districts

Kara77779 opened the pull request:
## Redistricting requirements
In Indiana, we consult [NCSL Redistricting Law 2020](https://documents.ncsl.org/wwwncsl/Redistricting-Census/Redistricting-Law-2020_NCSL%20FINAL.pdf) and impose the following constraints. In our simulations, legislative districts must:

1. Be contiguous [NCSL, 184]
2. Equal population [NCSL, 23]

### Algorithmic Constraints
We enforce a maximum population deviation of 5.0%.

## Data Sources
Data for Indiana comes from the ALARM Project's [2020 Redistricting Data Files](https://alarm-redist.github.io/posts/2021-08-10-census-2020/).

## Pre-processing Notes
No manual pre-processing decisions were necessary.

## Simulation Notes
We sample 10,000 districting plans for Indiana's lower house across 5 independent runs of the SMC algorithm.
No special techniques were needed to produce the sample.

We sample 10,000 districting plans for Indiana's upper house across 5 independent runs of the SMC algorithm.
No special techniques were needed to produce the sample.


## Validation
### State House Districts:
<img width="3000" height="5400" alt="validation_20260109_2308" src="https://github.com/user-attachments/assets/2a4582fa-d99f-481c-a1a8-6b990c3ea62b" />

```
SMC with Merge-Split MCMC Steps: 10,000 sampled plans of 100 districts on 5,159 units
Plans sampled on Linking Edge Forest Space using the Uniform Valid Edge forward kernel.
SMC Parameters:
• `weight_type` = optimal
• `seq_alpha` = 1
• `pop_temper` = 0
Mergesplit Parameters:
• `total_ms_steps` = 99
• `mh_accept_per_smc` = 49
• `pair_rule` = uniform
Plan diversity 80% range: 0.69 to 0.77
68 rhat values are `NA`
Largest R-hat values for ordered summary statistics:
        adv_16         adv_18         adv_20         arv_16         arv_18         arv_20 atg_16_dem_arr atg_16_rep_hil 
         1.031          1.027          1.015          1.019          1.020          1.018          1.032          1.018 
atg_20_dem_wei atg_20_rep_rok      comp_edge    comp_polsby  county_splits          e_dem          e_dvs           egap 
         1.014          1.017          1.007          1.014          1.009          1.005          1.041          1.005 
gov_16_dem_gre gov_16_rep_hol gov_20_dem_mye gov_20_rep_hol    muni_splits        ndshare            ndv            nrv 
         1.027          1.019          1.025          1.017          1.016          1.042          1.030          1.024 
         pbias       plan_dev       pop_aian      pop_asian      pop_black       pop_hisp       pop_nhpi      pop_other 
         1.005          1.000          1.011          1.017          1.048          1.043          1.021          1.032 
   pop_overlap        pop_two      pop_white         pr_dem pre_16_dem_cli pre_16_rep_tru pre_20_dem_bid pre_20_rep_tru 
         1.033          1.015          1.016          1.015          1.027          1.018          1.021          1.019 
sos_18_dem_har sos_18_rep_law      total_vap uss_16_dem_bay uss_16_rep_you uss_18_dem_don uss_18_rep_bra       vap_aian 
         1.029          1.019          1.005          1.028          1.023          1.025          1.022          1.017 
     vap_asian      vap_black       vap_hisp       vap_nhpi      vap_other        vap_two      vap_white 
         1.021          1.047          1.034          1.018          1.026          1.014          1.017 
• R-hat ≤ 1.05: 5432
• 1.05 < R-hat ≤ 1.1: 0
• R-hat > 1.1: 0
• Total R-hats: 5432
Sampling diagnostics for SMC run 1 of 5 (10,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd  Max. unique 
Split 1       891 (44.5%)    100.0%        1.29 1,249 ( 99%) 
Split 2       973 (48.7%)     99.6%        1.21   943 ( 75%) 
Split 3     1,123 (56.2%)     98.5%        1.06   986 ( 78%) 
Split 4     1,173 (58.7%)     97.8%        0.98 1,057 ( 84%) 
Split 5     1,234 (61.7%)     97.8%        0.94 1,063 ( 84%) 
Split 6     1,275 (63.7%)     96.2%        0.90 1,072 ( 85%) 
Split 7     1,334 (66.7%)     95.6%        0.84 1,091 ( 86%) 
Split 8     1,364 (68.2%)     94.3%        0.80 1,092 ( 86%) 
Split 9     1,444 (72.2%)     93.6%        0.72 1,117 ( 88%) 
Split 10    1,459 (73.0%)     92.8%        0.69 1,132 ( 90%) 
Split 11    1,465 (73.3%)     91.8%        0.67 1,156 ( 91%) 
Split 12    1,491 (74.5%)     90.8%        0.64 1,154 ( 91%) 
Split 13    1,528 (76.4%)     89.6%        0.63 1,167 ( 92%) 
Split 14    1,548 (77.4%)     88.4%        0.61 1,143 ( 90%) 
Split 15    1,575 (78.7%)     88.0%        0.59 1,154 ( 91%) 
Split 16    1,620 (81.0%)     86.7%        0.55 1,150 ( 91%) 
Split 17    1,636 (81.8%)     86.4%        0.53 1,180 ( 93%) 
Split 18    1,646 (82.3%)     86.2%        0.52 1,165 ( 92%) 
Split 19    1,618 (80.9%)     85.0%        0.53 1,221 ( 97%) 
Split 20    1,669 (83.5%)     82.7%        0.49 1,186 ( 94%) 
Split 21    1,680 (84.0%)     83.4%        0.49 1,203 ( 95%) 
Split 22    1,706 (85.3%)     81.7%        0.45 1,190 ( 94%) 
Split 23    1,704 (85.2%)     79.6%        0.45 1,170 ( 93%) 
Split 24    1,722 (86.1%)     79.2%        0.43 1,204 ( 95%) 
Split 25    1,737 (86.8%)     78.6%        0.43 1,196 ( 95%) 
Split 26    1,753 (87.6%)     77.2%        0.43 1,223 ( 97%) 
Split 27    1,751 (87.5%)     76.7%        0.41 1,214 ( 96%) 
Split 28    1,775 (88.7%)     75.7%        0.39 1,240 ( 98%) 
Split 29    1,776 (88.8%)     74.7%        0.38 1,212 ( 96%) 
Split 30    1,783 (89.1%)     75.6%        0.37 1,237 ( 98%) 
Split 31    1,798 (89.9%)     74.0%        0.36 1,243 ( 98%) 
Split 32    1,794 (89.7%)     73.8%        0.37 1,208 ( 96%) 
Split 33    1,806 (90.3%)     70.6%        0.35 1,243 ( 98%) 
Split 34    1,798 (89.9%)     70.8%        0.36 1,221 ( 97%) 
Split 35    1,813 (90.6%)     72.0%        0.35 1,238 ( 98%) 
Split 36    1,816 (90.8%)     69.4%        0.34 1,204 ( 95%) 
Split 37    1,822 (91.1%)     67.7%        0.33 1,225 ( 97%) 
Split 38    1,832 (91.6%)     68.4%        0.32 1,253 ( 99%) 
Split 39    1,830 (91.5%)     67.4%        0.32 1,240 ( 98%) 
Split 40    1,843 (92.2%)     67.4%        0.31 1,219 ( 96%) 
Split 41    1,850 (92.5%)     65.2%        0.30 1,233 ( 98%) 
Split 42    1,848 (92.4%)     65.7%        0.30 1,237 ( 98%) 
Split 43    1,850 (92.5%)     63.4%        0.30 1,234 ( 98%) 
Split 44    1,852 (92.6%)     63.5%        0.30 1,231 ( 97%) 
Split 45    1,859 (92.9%)     63.7%        0.29 1,227 ( 97%) 
Split 46    1,869 (93.4%)     61.0%        0.27 1,225 ( 97%) 
Split 47    1,872 (93.6%)     61.0%        0.27 1,230 ( 97%) 
Split 48    1,880 (94.0%)     59.0%        0.26 1,202 ( 95%) 
Split 49    1,881 (94.1%)     60.4%        0.26 1,230 ( 97%) 
Split 50    1,886 (94.3%)     57.5%        0.26 1,228 ( 97%) 
Split 51    1,888 (94.4%)     57.5%        0.25 1,240 ( 98%) 
Split 52    1,882 (94.1%)     57.5%        0.26 1,229 ( 97%) 
Split 53    1,894 (94.7%)     56.7%        0.24 1,230 ( 97%) 
Split 54    1,892 (94.6%)     55.8%        0.24 1,245 ( 98%) 
Split 55    1,899 (94.9%)     56.2%        0.24 1,264 (100%) 
Split 56    1,897 (94.8%)     55.8%        0.24 1,242 ( 98%) 
Split 57    1,901 (95.1%)     55.3%        0.23 1,243 ( 98%) 
Split 58    1,895 (94.8%)     52.2%        0.24 1,251 ( 99%) 
Split 59    1,902 (95.1%)     53.1%        0.24 1,220 ( 97%) 
Split 60    1,912 (95.6%)     51.8%        0.22 1,257 ( 99%) 
Split 61    1,913 (95.7%)     51.0%        0.22 1,236 ( 98%) 
Split 62    1,917 (95.8%)     51.2%        0.22 1,241 ( 98%) 
Split 63    1,920 (96.0%)     49.4%        0.21 1,252 ( 99%) 
Split 64    1,924 (96.2%)     50.1%        0.20 1,271 (101%) 
Split 65    1,921 (96.0%)     49.2%        0.21 1,233 ( 98%) 
Split 66    1,924 (96.2%)     48.6%        0.20 1,229 ( 97%) 
Split 67    1,932 (96.6%)     48.8%        0.19 1,238 ( 98%) 
Split 68    1,932 (96.6%)     46.8%        0.19 1,249 ( 99%) 
Split 69    1,930 (96.5%)     46.5%        0.19 1,249 ( 99%) 
Split 70    1,930 (96.5%)     45.5%        0.20 1,239 ( 98%) 
Split 71    1,931 (96.5%)     46.1%        0.19 1,225 ( 97%) 
Split 72    1,935 (96.8%)     46.5%        0.19 1,238 ( 98%) 
Split 73    1,937 (96.8%)     44.5%        0.18 1,235 ( 98%) 
Split 74    1,936 (96.8%)     43.4%        0.19 1,235 ( 98%) 
Split 75    1,942 (97.1%)     44.0%        0.17 1,224 ( 97%) 
Split 76    1,939 (97.0%)     43.2%        0.18 1,246 ( 99%) 
Split 77    1,942 (97.1%)     42.4%        0.18 1,218 ( 96%) 
Split 78    1,947 (97.4%)     42.3%        0.17 1,252 ( 99%) 
Split 79    1,947 (97.4%)     41.4%        0.17 1,250 ( 99%) 
Split 80    1,947 (97.3%)     40.1%        0.17 1,241 ( 98%) 
Split 81    1,950 (97.5%)     40.9%        0.16 1,264 (100%) 
Split 82    1,947 (97.3%)     40.4%        0.17 1,274 (101%) 
Split 83    1,956 (97.8%)     39.1%        0.15 1,240 ( 98%) 
Split 84    1,956 (97.8%)     39.7%        0.15 1,246 ( 99%) 
Split 85    1,955 (97.7%)     38.1%        0.15 1,239 ( 98%) 
Split 86    1,954 (97.7%)     38.8%        0.16 1,238 ( 98%) 
Split 87    1,958 (97.9%)     37.5%        0.15 1,243 ( 98%) 
Split 88    1,960 (98.0%)     35.9%        0.15 1,252 ( 99%) 
Split 89    1,957 (97.8%)     36.1%        0.15 1,253 ( 99%) 
Split 90    1,961 (98.0%)     35.8%        0.14 1,240 ( 98%) 
Split 91    1,961 (98.1%)     34.5%        0.14 1,228 ( 97%) 
Split 92    1,963 (98.1%)     33.7%        0.14 1,247 ( 99%) 
Split 93    1,965 (98.2%)     33.6%        0.14 1,271 (101%) 
Split 94    1,966 (98.3%)     32.4%        0.13 1,208 ( 96%) 
Split 95    1,967 (98.4%)     33.2%        0.13 1,202 ( 95%) 
Split 96    1,969 (98.5%)     32.5%        0.13 1,203 ( 95%) 
Split 97    1,974 (98.7%)     30.8%        0.12 1,218 ( 96%) 
Split 98    1,976 (98.8%)     31.3%        0.11 1,160 ( 92%) 
Split 99    1,979 (98.9%)     30.0%        0.11 1,101 ( 87%) 
Resample    1,979 (98.9%)       NA%          NA 1,922 (152%) 

Sampling diagnostics for Mergesplit Steps of SMC run 1 of 5 (10,000 samples)
           Acc. rate MS Steps per Plan
MS Step 1      47.4%                49
MS Step 2      47.3%               147
MS Step 3      47.6%               147
MS Step 4      47.8%               147
MS Step 5      47.8%               147
MS Step 6      47.9%               147
MS Step 7      48.2%               147
MS Step 8      48.3%               147
MS Step 9      48.2%               147
MS Step 10     48.2%               147
MS Step 11     48.1%               147
MS Step 12     48.2%               147
MS Step 13     48.0%               147
MS Step 14     48.1%               147
MS Step 15     48.2%               147
MS Step 16     47.8%               147
MS Step 17     47.8%               147
MS Step 18     47.9%               147
MS Step 19     47.4%               147
MS Step 20     47.3%               147
MS Step 21     47.6%               147
MS Step 22     47.4%               147
MS Step 23     47.1%               147
MS Step 24     47.1%               147
MS Step 25     46.9%               147
MS Step 26     46.8%               147
MS Step 27     46.4%               147
MS Step 28     46.2%               147
MS Step 29     46.2%               147
MS Step 30     46.2%               147
MS Step 31     46.0%               147
MS Step 32     45.7%               147
MS Step 33     45.5%               147
MS Step 34     45.3%               147
MS Step 35     45.3%               147
MS Step 36     45.2%               147
MS Step 37     45.0%               147
MS Step 38     44.6%               147
MS Step 39     44.0%               147
MS Step 40     44.2%               147
MS Step 41     44.1%               147
MS Step 42     43.9%               147
MS Step 43     43.9%               147
MS Step 44     43.5%               147
MS Step 45     43.5%               147
MS Step 46     43.3%               147
MS Step 47     43.1%               147
MS Step 48     42.9%               147
MS Step 49     42.9%               147
MS Step 50     42.5%               147
MS Step 51     42.3%               147
MS Step 52     42.1%               147
MS Step 53     41.7%               147
MS Step 54     41.7%               147
MS Step 55     41.7%               147
MS Step 56     41.4%               147
MS Step 57     41.2%               147
MS Step 58     41.1%               147
MS Step 59     40.8%               147
MS Step 60     40.6%               147
MS Step 61     40.3%               147
MS Step 62     40.1%               147
MS Step 63     40.0%               147
MS Step 64     39.8%               147
MS Step 65     39.5%               147
MS Step 66     39.5%               147
MS Step 67     39.2%               147
MS Step 68     38.9%               147
MS Step 69     38.7%               147
MS Step 70     38.4%               147
MS Step 71     38.4%               147
MS Step 72     37.9%               147
MS Step 73     38.0%               147
MS Step 74     37.6%               147
MS Step 75     37.6%               147
MS Step 76     37.2%               147
MS Step 77     36.9%               147
MS Step 78     36.7%               147
MS Step 79     36.4%               147
MS Step 80     36.3%               147
MS Step 81     36.1%               147
MS Step 82     35.9%               147
MS Step 83     35.6%               147
MS Step 84     35.5%               147
MS Step 85     35.2%               147
MS Step 86     35.1%               147
MS Step 87     34.8%               147
MS Step 88     34.6%               147
MS Step 89     34.4%               147
MS Step 90     34.1%               147
MS Step 91     34.0%               147
MS Step 92     33.5%               147
MS Step 93     33.4%               147
MS Step 94     33.1%               147
MS Step 95     32.9%               196
MS Step 96     32.7%               196
MS Step 97     32.3%               196
MS Step 98     32.1%               196
MS Step 99     31.9%               196

•  Watch out for low effective samples, very low acceptance rates (less than 1%), large std. devs. of the log weights
(more than 3 or so), and low numbers of unique plans. R-hat values for summary statistics should be between 1 and 1.05.
```
### State Senate Districts:

 State Senate Districts:
<img width="3000" height="5400" alt="validation_20260109_2312" src="https://github.com/user-attachments/assets/fd6ade9d-c1e9-408a-b145-855fb68b3320" />

```
SMC with Merge-Split MCMC Steps: 10,000 sampled plans of 50 districts on 5,159 units
Plans sampled on Linking Edge Forest Space using the Uniform Valid Edge forward kernel.

SMC Parameters:
• `weight_type` = optimal
• `seq_alpha` = 1
• `pop_temper` = 0
Mergesplit Parameters:
• `total_ms_steps` = 49
• `mh_accept_per_smc` = 42
• `pair_rule` = uniform
Plan diversity 80% range: 0.68 to 0.79
33 rhat values are `NA`

Largest R-hat values for ordered summary statistics:
        adv_16         adv_18         adv_20         arv_16         arv_18         arv_20 atg_16_dem_arr atg_16_rep_hil 
         1.010          1.016          1.017          1.006          1.005          1.008          1.015          1.005 
atg_20_dem_wei atg_20_rep_rok      comp_edge    comp_polsby  county_splits          e_dem          e_dvs           egap 
         1.015          1.007          1.004          1.010          1.001          1.003          1.011          1.003 
gov_16_dem_gre gov_16_rep_hol gov_20_dem_mye gov_20_rep_hol    muni_splits        ndshare            ndv            nrv 
         1.006          1.006          1.018          1.008          1.012          1.011          1.019          1.007 
         pbias       plan_dev       pop_aian      pop_asian      pop_black       pop_hisp       pop_nhpi      pop_other 
         1.002          1.000          1.019          1.011          1.010          1.017          1.008          1.016 
   pop_overlap        pop_two      pop_white         pr_dem pre_16_dem_cli pre_16_rep_tru pre_20_dem_bid pre_20_rep_tru 
         1.020          1.021          1.004          1.004          1.012          1.004          1.011          1.008 
sos_18_dem_har sos_18_rep_law      total_vap uss_16_dem_bay uss_16_rep_you uss_18_dem_don uss_18_rep_bra       vap_aian 
         1.016          1.005          1.004          1.010          1.006          1.020          1.007          1.020 
     vap_asian      vap_black       vap_hisp       vap_nhpi      vap_other        vap_two      vap_white 
         1.008          1.010          1.020          1.014          1.011          1.009          1.004 

• R-hat ≤ 1.05: 2717
• 1.05 < R-hat ≤ 1.1: 0
• R-hat > 1.1: 0
• Total R-hats: 2717
Sampling diagnostics for SMC run 1 of 5 (10,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd  Max. unique 
Split 1       836 (41.8%)    100.0%        1.23 1,253 ( 99%) 
Split 2       996 (49.8%)     98.7%        1.21   947 ( 75%) 
Split 3     1,075 (53.8%)     96.7%        1.10   997 ( 79%) 
Split 4     1,123 (56.2%)     96.1%        1.01 1,007 ( 80%) 
Split 5     1,215 (60.8%)     94.3%        0.92 1,032 ( 82%) 
Split 6     1,303 (65.2%)     91.7%        0.84 1,091 ( 86%) 
Split 7     1,311 (65.5%)     90.6%        0.84 1,099 ( 87%) 
Split 8     1,376 (68.8%)     88.6%        0.77 1,077 ( 85%) 
Split 9     1,426 (71.3%)     86.8%        0.72 1,106 ( 87%) 
Split 10    1,420 (71.0%)     85.2%        0.71 1,138 ( 90%) 
Split 11    1,473 (73.7%)     82.5%        0.64 1,145 ( 91%) 
Split 12    1,489 (74.5%)     81.4%        0.65 1,154 ( 91%) 
Split 13    1,519 (76.0%)     78.2%        0.62 1,145 ( 91%) 
Split 14    1,567 (78.4%)     77.6%        0.60 1,150 ( 91%) 
Split 15    1,577 (78.9%)     78.4%        0.56 1,135 ( 90%) 
Split 16    1,582 (79.1%)     75.1%        0.56 1,189 ( 94%) 
Split 17    1,620 (81.0%)     72.0%        0.53 1,178 ( 93%) 
Split 18    1,636 (81.8%)     71.8%        0.51 1,187 ( 94%) 
Split 19    1,642 (82.1%)     71.5%        0.49 1,159 ( 92%) 
Split 20    1,655 (82.7%)     66.0%        0.49 1,192 ( 94%) 
Split 21    1,668 (83.4%)     66.9%        0.47 1,171 ( 93%) 
Split 22    1,672 (83.6%)     64.9%        0.45 1,177 ( 93%) 
Split 23    1,715 (85.8%)     64.1%        0.43 1,186 ( 94%) 
Split 24    1,713 (85.7%)     62.6%        0.43 1,199 ( 95%) 
Split 25    1,722 (86.1%)     61.6%        0.42 1,199 ( 95%) 
Split 26    1,726 (86.3%)     60.7%        0.43 1,193 ( 94%) 
Split 27    1,750 (87.5%)     59.4%        0.41 1,217 ( 96%) 
Split 28    1,761 (88.0%)     57.8%        0.39 1,208 ( 96%) 
Split 29    1,782 (89.1%)     55.4%        0.37 1,210 ( 96%) 
Split 30    1,772 (88.6%)     55.6%        0.37 1,211 ( 96%) 
Split 31    1,779 (89.0%)     54.9%        0.37 1,198 ( 95%) 
Split 32    1,797 (89.8%)     53.8%        0.35 1,211 ( 96%) 
Split 33    1,801 (90.0%)     53.2%        0.34 1,243 ( 98%) 
Split 34    1,813 (90.6%)     50.5%        0.33 1,219 ( 96%) 
Split 35    1,824 (91.2%)     50.5%        0.32 1,195 ( 95%) 
Split 36    1,843 (92.2%)     50.9%        0.30 1,207 ( 95%) 
Split 37    1,842 (92.1%)     49.2%        0.30 1,225 ( 97%) 
Split 38    1,858 (92.9%)     46.7%        0.28 1,185 ( 94%) 
Split 39    1,852 (92.6%)     46.3%        0.29 1,217 ( 96%) 
Split 40    1,842 (92.1%)     44.4%        0.30 1,219 ( 96%) 
Split 41    1,864 (93.2%)     43.4%        0.28 1,218 ( 96%) 
Split 42    1,874 (93.7%)     43.2%        0.26 1,197 ( 95%) 
Split 43    1,874 (93.7%)     41.8%        0.26 1,231 ( 97%) 
Split 44    1,882 (94.1%)     41.3%        0.25 1,182 ( 93%) 
Split 45    1,899 (94.9%)     40.6%        0.24 1,222 ( 97%) 
Split 46    1,904 (95.2%)     38.6%        0.23 1,222 ( 97%) 
Split 47    1,916 (95.8%)     37.8%        0.22 1,185 ( 94%) 
Split 48    1,926 (96.3%)     35.7%        0.20 1,202 ( 95%) 
Split 49    1,930 (96.5%)     34.0%        0.19 1,096 ( 87%) 
Resample    1,930 (96.5%)       NA%          NA 1,850 (146%) 

Sampling diagnostics for Mergesplit Steps of SMC run 1 of 5 (10,000 samples)
           Acc. rate MS Steps per Plan
MS Step 1      47.3%                42
MS Step 2      46.5%               126
MS Step 3      46.8%               126
MS Step 4      46.7%               126
MS Step 5      46.8%               126
MS Step 6      46.4%               126
MS Step 7      46.4%               126
MS Step 8      45.9%               126
MS Step 9      45.6%               126
MS Step 10     45.3%               126
MS Step 11     44.8%               126
MS Step 12     44.2%               126
MS Step 13     44.2%               126
MS Step 14     43.8%               126
MS Step 15     43.5%               126
MS Step 16     43.4%               126
MS Step 17     42.6%               126
MS Step 18     42.4%               126
MS Step 19     42.1%               126
MS Step 20     42.1%               126
MS Step 21     41.9%               126
MS Step 22     40.7%               126
MS Step 23     40.5%               126
MS Step 24     40.0%               126
MS Step 25     39.8%               126
MS Step 26     39.5%               126
MS Step 27     39.0%               126
MS Step 28     38.9%               126
MS Step 29     38.4%               126
MS Step 30     38.2%               126
MS Step 31     37.6%               126
MS Step 32     37.3%               126
MS Step 33     37.1%               126
MS Step 34     36.6%               126
MS Step 35     36.4%               126
MS Step 36     35.9%               126
MS Step 37     35.7%               126
MS Step 38     35.5%               126
MS Step 39     35.0%               126
MS Step 40     34.9%               126
MS Step 41     34.2%               126
MS Step 42     33.8%               126
MS Step 43     33.6%               126
MS Step 44     32.9%               126
MS Step 45     32.6%               168
MS Step 46     32.0%               168
MS Step 47     31.5%               168
MS Step 48     31.5%               168
MS Step 49     30.9%               168

•  Watch out for low effective samples, very low acceptance rates (less than 1%), large std. devs. of the log weights
(more than 3 or so), and low numbers of unique plans. R-hat values for summary statistics should be between 1 and 1.05.
```
## Checklist

- [x] I have followed the [instructions](https://github.com/alarm-redist/fifty-states/blob/main/CONTRIBUTING.md)
- [x] I have updated the [tracker](https://docs.google.com/spreadsheets/d/1nV1vrzQZBCil9MdQV_8AOQr4Y4X2evYO9R4WeZkiO3g/edit?gid=0#gid=0)
- [x] All `TODO` lines from the template code have been removed
- [x] I have merged in the main branch and then recalculated summary statistics
- [x] I have run `enforce_style()` to format my code
- [x] The documentation copied above is up-to-date 
- [x] There are no data files in this pull request
- [x] None of the file output paths (for the `redist_map` and `redist_plans` objects, and summary statistics) have been edited

@philipwosull @ethanjasny

Kara77779 committed:
Add IN-leg-2020

philipwosull commented:
Could you recheck these citations/page numbers? Some of them are off (I'm not sure if you used the page numbers in the document itself or the pdf file but please make sure to use the page numbers listed on the pages of the document itself.)

The page numbers for 1,3, and 4 should be 184 (although it seems like Indiana actually does not mandate compactness or preserving county boundaries.) Also Davis v. Bandemer is not relevant anymore (See Rucho v. Common Cause holding "partisan gerrymandering claims present political questions beyond the reach of the federal courts").

Kara77779 commented:
> Could you recheck these citations/page numbers? Some of them are off (I'm not sure if you used the page numbers in the document itself or the pdf file but please make sure to use the page numbers listed on the pages of the document itself.)
> 
> The page numbers for 1,3, and 4 should be 184 (although it seems like Indiana actually does not mandate compactness or preserving county boundaries.) Also Davis v. Bandemer is not relevant anymore (See Rucho v. Common Cause holding "partisan gerrymandering claims present political questions beyond the reach of the federal courts").

Thanks for the feedback earlier!
I’ve rechecked all page references using the printed pagination in NCSL Redistricting Law 2020 (not the PDF page count).
Updated Indiana to cite p. 184 (Appendix D) for contiguity and other traditional geographic criteria.
Noted that compactness and county/municipal boundary preservation are not statutory requirements in Indiana, per Appendix D.
Cited p. 23 for the equal-population rule (14th Amendment, “one person, one vote”).
Replaced the outdated Bandemer reference with Rucho v. Common Cause (2019, p. 106).
I’ll next revise Kentucky and Alabama using the same approach—verifying each page number against the printed pagination and confirming whether each criterion is statutory or traditional.
Please let me know if this aligns with the expected citation format.

philipwosull commented:
@Kara77779 The state house looks good! For the state senate the total county splits look a little high. Could you try adding a mild total splits constraint and seeing if it helps. (Just run it with a small value of like 2 or 3 for `mh_accept_per_smc` to see how it looks, no need to do a big expensive rerun yet). If that doesn't work let me know and we'll try something else.

Kara77779 commented:
> @Kara77779 The state house looks good! For the state senate the total county splits look a little high. Could you try adding a mild total splits constraint and seeing if it helps. (Just run it with a small value of like 2 or 3 for `mh_accept_per_smc` to see how it looks, no need to do a big expensive rerun yet). If that doesn't work let me know and we'll try something else.


Hi Philip, thanks for the feedback!
For the IN state senate, I'm planning to:

Add a total splits constraint:

rconstr <- redist_constr(map_ssd) %>%
    add_constr_total_splits(strength = 2, admin = county)

Run a quick test with small sample size (nsims = 500, runs = 1) and mh_accept_per_smc = 3
Check if the total_county_splits distribution moves closer to the enacted plan

Does this look right to you?

philipwosull commented:
@Kara77779 Could you regenerate these plots with the updated diagnostic code? This one is missing the total splits and reock compactness plots.
